### PR TITLE
Feature/calendar/time logic

### PR DESCRIPTION
**Pull request by patrickb42**
# Short Description

Allows students to be checked in at least a half-hour before their session.

## Rationale/purpose of the feature

This ensures students can't be mistakenly checked in by ensuring that it's at least a half-hour before the scheduled session. Time lock automatically updates so the mentor can actively see meeting statuses.

## Link to relevant User Story

[add logic for check-in times (30 minutes before session/expired, etc)](https://trello.com/c/pq04d9lx)

## Checklist for author and reviewers

- [ ] Commits are well named and descriptive
- [ ] PR is focused on a single problem, is small enough and easy to review
- [ ] PR requests were sent to multiple reviewers
- [ ] PR is reviewed by more than one team member
- [ ] Code is consistently formatted
- [ ] Code is suitably commented on
- [ ] There is no duplicate code
- [ ] Code tries to follow best practices
- [ ] Decide with reviewers if this PR should have tests before merging

### CODE REVIEW

File: src/components/Calendar/CheckinModal.js [hunk 1]
```diff
@@ -1,79 +1,121 @@
...
-        <strong>Computer: </strong>
-        {props.details.resourceId}
-      </p>
-      <Switch
-        checked={checkedIn}
-        checkedChildren="Student Checked In"
-        unCheckedChildren="Check-in Student"
-        onChange={handleCheckIn}
-      />
+      <h6>ExactStatus: {status}</h6>
```
Comment (petevs) [inline src/components/Calendar/CheckinModal.js:90]: I don't know if we need exact status and status in the check-in modal. Maybe we just pick one? The exact status isn't written in friendly though, so if we go with that maybe we remove the underscores.

File: src/components/Calendar/EventListSideBar.js [hunk 2]
```diff
@@ -1,84 +1,46 @@
...
+      event={event}
+      forceRender={forceRender}
+    />
   );
 
   return (
     <div className="sidebar-container">
-      <div className="calendar-container">
-        <div className="rbc-toolbar rbc-btn-group">
-          <Dropdown overlay={schoolMenu}>
```
Comment (petevs) [inline src/components/Calendar/EventListSideBar.js:51]: We need to have the dropdown, so we can filter the events. Carlos' and other future pull requests are building off this feature, so we may run into some issues merging.

**Review by petevs — COMMENTED** [review]
Functionality seems to be working great. Two things I noticed were the double status on the checkin modal and the missing location dropdown on the sidebar.

**Review by rmscrlos — COMMENTED** [review]
Functionality works great! Thank you so much for making this work.  Great work!

**Review by rmscrlos — APPROVED** [review]

File: src/components/Calendar/EventListSideBar.js (hunk 2 above)
Comment (patrickb42) [reply src/components/Calendar/EventListSideBar.js]: added back in

File: src/components/Calendar/CheckinModal.js (hunk 1 above)
Comment (patrickb42) [reply src/components/Calendar/CheckinModal.js]: it's debug code; I've removed it.

File: src/components/Calendar/EventListSideBar.js (hunk 2 above)
Comment (petevs) [reply src/components/Calendar/EventListSideBar.js]: Great, Thanks! Should be good now for the other merges!

**Review by petevs — APPROVED** [review]

**Pull request merged**